dev/release/post-01-merge.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -e
+set -u
+
+if [ "$#" -ne 2 ]; then
+  echo "Usage: $0 <version> <rc-num>"
+  exit
+fi
+
+version=$1
+rc=$2
+
+case ${version} in
+  *.*.0) # Major release
+    echo "Open a pull request:"
+    echo "  https://github.com/apache/arrow/compare/master...${USER}:release-${version}-rc${rc}?expand=1"
+    echo
+    echo "Merge the pull request and tag the merged commit:"
+    echo "  git fetch --all --prune --tags --force -j\$(nproc)"
+    echo "  git checkout master"
+    echo "  git rebase apache/master"
+    echo "  git tag -a -m 'Apache Arrow ${version}' apache-arrow-${version}"
```

Comment:
The `apache-arrow-${version}` tag should point to the `release-${version}-rc${rc}` branch since the master possibly receives new commits during the release verification. 

On the other hand we need to place a `apache-arrow-${next_version}.dev` commit to keep the SCM versioning tools (e.g. setuptools-scm) functional.

Reply:
> The `apache-arrow-${version}` tag should point to the `release-${version}-rc${rc}` branch since the master possibly receives new commits during the release verification. 

It's `release-${version}` not `release-${version}-rc${rc}`, right?

Generally, it seems that we don't push `release-${version}-rc${rc}` to apache/arrow: https://github.com/apache/arrow/branches/all?query=release-

Reply:
Right. I've just deleted the `release-6.0.1-rc0` branch I accidentally pushed.